### PR TITLE
nnvp-77

### PR DESCRIPTION
**Pull request by eryk7b**

### CODE REVIEW

**Comment by laurent-theglue** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-39** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-39.cluster.development.nnvp.thegluesandbox.com)

File: src/main/java/com/theglue/nnvp/controller/validator/GetVehiclesTypeDetailsByTypeRequestValidator.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  public void validate(GetVehicleTypeDetailsByTypeRequestDTO request) {
+    ValidationUtils.checkCondition(
+        request.getLocalVehicleTypeIdentifier() != null
+            || Booleans.asList(request.getYear() != null,
+            StringUtils.isNotBlank(request.getMakeCode()),
+            StringUtils.isNotBlank(request.getModelCode()),
+            request.getTypeCode() != null)
+            .stream()
+            .allMatch(bool -> bool),
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/controller/validator/GetVehiclesTypeDetailsByTypeRequestValidator.java]: .allMatch(Function::identity)

File: src/main/java/com/theglue/nnvp/controller/dto/VehicleTypeDetailsDTO.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  private Integer weightMaximum;
+
+  private Integer numberOfSeats;
+
+  private Segment segment;
+
+  private FuelType fuelType;
+
+  // marked as see further but no enum CoupeIndicator found in documentation
+  private String coupeIndicator;
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/controller/dto/VehicleTypeDetailsDTO.java]: mark as TODO to not forget

**Comment by laurent-theglue** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-39** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-39.cluster.development.nnvp.thegluesandbox.com)

**Pull request merged**